src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +43,36 @@
      * Андрей - 600
      * София - 900</p>
      */
+    static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("d.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+            throws IllegalDateParametersException {
+        if (LocalDate.parse(dateFrom, DATE_FORMAT).isAfter(LocalDate.parse(dateTo, DATE_FORMAT))) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+        StringBuilder report = new StringBuilder();
+        report.append("Отчёт за период ").append(dateFrom).append(" - ")
+                .append(dateTo).append("\n");
+        int totalSalaryForThePeriod = 0;
```

Comment:
declare variable inside of the loop, and you won't need to refresh in manually

Reply:
done